src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -1020,6 +1023,129 @@ void MeshForwarder::UpdateRoutes(uint8_t *           aFrame,
     return;
 }
 
+void MeshForwarder::UpdateFragmentLifetime(void)
+{
+    bool shouldRun = false;
+
+    for (size_t i = 0; i < OT_ARRAY_LENGTH(mFragmentEntries); i++)
+    {
+        if (mFragmentEntries[i].GetLifetime() != 0)
+        {
+            mFragmentEntries[i].SetLifetime(mFragmentEntries[i].GetLifetime() - 1);
+
+            if (mFragmentEntries[i].GetLifetime() != 0)
+            {
+                shouldRun = true;
+            }
+        }
+    }
+
+    if (shouldRun && !mUpdateTimer.IsRunning())
+    {
+        mUpdateTimer.Start(kStateUpdatePeriod);
+    }
+}
+
+void MeshForwarder::UpdateFragmentPriority(Lowpan::FragmentHeader &aFragmentHeader,
+                                           uint8_t                 aFragmentLength,
+                                           uint8_t                 aPriority)
+{
+    FragmentPriorityEntry *entry;
+
+    if (aFragmentHeader.GetDatagramOffset() == 0)
+    {
+        VerifyOrExit((entry = GetFragmentPriorityEntry(false)) != NULL);
+
+        entry->SetDatagramTag(aFragmentHeader.GetDatagramTag());
+        entry->SetPriority(aPriority);
+        entry->SetLifetime(kReassemblyTimeout);
+
+        if (!mUpdateTimer.IsRunning())
+        {
+            mUpdateTimer.Start(kStateUpdatePeriod);
+        }
+    }
+    else
+    {
+        VerifyOrExit((entry = GetFragmentPriorityEntry(true, aFragmentHeader.GetDatagramTag())) != NULL);
+
+        if (aFragmentHeader.GetDatagramOffset() + aFragmentLength >= aFragmentHeader.GetDatagramSize())
+        {
+            entry->SetLifetime(0);
+        }
+    }
+
+exit:
+    return;
+}
+
+FragmentPriorityEntry *MeshForwarder::GetFragmentPriorityEntry(bool aValid, uint16_t aTag)
```

Comment:
Suggestion: how about breaking this into two methods instead to help make the code read better?
- `FindFragmentPriorityEntry(uint16_t aTag)` --> Finds an entry for a given tag (if one exists)
- `GetUnusedFragementPriorityEntry()` --> Gets an unused entry.

Reply:
Changed.